test/test_tensor_creation_ops.py

```diff
@@ -832,6 +832,86 @@ def compare_helper_(like_fn, t):
                     tp = t.permute(p)
                     compare_helper_(like_fn, tp)
 
+    def _hvd_split_helper(self, inputs, device, dtype, dim):
+        operator_names = ["torch.vsplit", "torch.hsplit", "torch.dsplit"]
+        operations_torch = [torch.vsplit, torch.hsplit, torch.dsplit]
+        operations_numpy = [np.vsplit, np.hsplit, np.dsplit]
+
+        dimension_error_message = operator_names[dim] + " requires a tensor with at least "
+        divisibiliy_error_message = operator_names[dim] + " attempted to split along dimension "
+        op_torch = operations_torch[dim]
+        op_numpy = operations_numpy[dim]
+
+        for shape, arg in inputs:
+            direction = dim - (len(shape) == 1 and dim == 1)
+            bound = dim + 2 * (dim == 0) + (dim == 2)
+            error_expected = len(shape) < bound or (not isinstance(arg, list) and shape[direction] % arg != 0)
+
+            t = make_tensor(shape, device, dtype)
+            t_np = t.cpu().numpy()
+
+            if not error_expected:
+                self.assertEqual(op_torch(t, arg), op_numpy(t_np, arg))
+            else:
+                self.assertRaises(RuntimeError, lambda: op_torch(t, arg))
+                self.assertRaises(ValueError, lambda: op_numpy(t, arg))
+                expected_error_message = dimension_error_message if len(shape) < bound else divisibiliy_error_message
+                self.assertRaisesRegex(RuntimeError, expected_error_message, lambda: op_torch(t, arg))
+
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_hsplit(self, device, dtype):
+        inputs = (
+            ((), 3),
+            ((), [2, 4, 6]),
+            ((6,), 2),
+            ((6,), 4),
+            ((6,), [2, 5]),
+            ((6,), [7, 9]),
+            ((3, 8), 4),
+            ((3, 8), 5),
+            ((3, 8), [1, 5]),
+            ((3, 8), [3, 8]),
+            ((5, 5, 5), 2),
+            ((5, 5, 5), [1, 4]),
+            ((5, 0, 5), 3),
+            ((5, 5, 0), [2, 6]),
+        )
+        self._hvd_split_helper(inputs, device, dtype, 1)
+
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
```

Comment:
This is very thorough, which is great, but let's just have these test long, float32, and complex64. That's more than sufficient to validate their logic.